src/main/java/org/jenkinsci/extension_indexer/SourceAndLibs.java

```diff
@@ -143,6 +143,8 @@ private static void downloadDependencies(File pomDir, File destDir) throws IOExc
             process = System.getenv("M2_HOME") + "/bin/mvn";
         }
         ProcessBuilder builder = new ProcessBuilder(process,
+                "--settings", new File("maven-settings.xml").getAbsolutePath(),
```

Comment:
BTW it is poor style to assume that the CWD of the process has a particular file in it. Rather out `maven-settings.xml` into `src/main/resources/` and use `FileUtils` to copy the resource to a local temp file.